internal/yunikorn/event_handler.go

```diff
@@ -234,3 +235,47 @@ func (s *Service) handleAppRemoveEvent(ctx context.Context, ev *si.EventRecord)
 			ev.GetEventChangeDetail())
 	}
 }
+
+func (s *Service) handleQueueEvent(ctx context.Context, ev *si.EventRecord) {
+	logger := log.FromContext(ctx)
+
+	switch ev.GetEventChangeType() {
+	case si.EventRecord_ADD:
+		s.queueAddAccumulator.add(ev)
+	case si.EventRecord_SET:
+		// Ignored for now
+	case si.EventRecord_REMOVE:
+		// Ignored for now
+	case si.EventRecord_NONE:
+	default:
+		// should be warning
+		logger.Warnf("unknown event EventChangeType for an Event of type QUEUE: %v", ev.GetEventChangeType())
+	}
+}
+
+func (s *Service) handleQueueEvents(ctx context.Context, events []*si.EventRecord) {
+	logger := log.FromContext(ctx)
+
+	logger.Debug("Processing queue events")
+	for _, event := range events {
+		logger.Debugf("Event: %v", event)
+	}
+
+	s.handleQueueAddEvent(ctx)
+	logger.Debug("Finished processing queue events")
+}
+
+func (s *Service) handleQueueAddEvent(ctx context.Context) {
+	logger := log.FromContext(ctx)
+
+	partitions, err := s.upsertPartitions(ctx)
+	if err != nil {
+		logger.Errorf("could not get partitions: %v", err)
+		return
+	}
+
+	if _, err := s.upsertPartitionQueues(ctx, partitions); err != nil {
```

Comment:
We will be removing `upsertPartitionQueues` as per https://github.com/G-Research/yunikorn-history-server/issues/143

A work in progress in https://github.com/G-Research/yunikorn-history-server/pull/221 to use the newly added CRUD.